docs/guidance/rayservice.md

```diff
@@ -0,0 +1,201 @@
+ray-operator-75dbbf8587-5lrvn   1/1     Running   0          31s
+```
+
+### Run an example cluster
+
+There is one example config file to deploy RaySerive included here:
+[ray_v1alpha1_rayservice.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray_v1alpha1_rayservice.yaml)
+
+```shell
+# Create a ray service and deploy fruit deployment graph.
+$ kubectl apply -f config/samples/ray_v1alpha1_rayservice.yaml
+```
+
+```shell
+# List running RayServices.
+$ kubectl get rayservice
+NAME                AGE
+rayservice-sample   7s
+```
+
+```shell
+# The created RayService should include a head pod, a worker pod, and four services.
+$ kubectl get pods
+NAME                                                      READY   STATUS    RESTARTS   AGE
+ervice-sample-raycluster-qd2vl-worker-small-group-bxpp6   1/1     Running   0          24m
+rayservice-sample-raycluster-qd2vl-head-45hj4             1/1     Running   0          24m
+
+$ kubectl get services
+NAME                                               TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                          AGE
+kubernetes                                         ClusterIP   10.100.0.1       <none>        443/TCP                                          62d
+# A head node service maintained by the RayService.
+rayservice-sample-head-svc                         ClusterIP   10.100.34.24     <none>        6379/TCP,8265/TCP,10001/TCP,8000/TCP,52365/TCP   24m
+# A dashboard agent service maintained by the RayCluster.
+rayservice-sample-raycluster-qd2vl-dashboard-svc   ClusterIP   10.100.109.177   <none>        52365/TCP                                        24m
+# A head node service maintained by the RayCluster.
+rayservice-sample-raycluster-qd2vl-head-svc        ClusterIP   10.100.180.221   <none>        6379/TCP,8265/TCP,10001/TCP,8000/TCP,52365/TCP   24m
+# A serve service maintained by the RayService.
+rayservice-sample-serve-svc                        ClusterIP   10.100.39.92     <none>        8000/TCP                                         24m
+```
+
+> Note: Default ports and their definition. 
+
+| Port  | Definition          |
+|-------|---------------------|
+| 6379  | Ray GCS             |
+| 8265  | Ray Dashboard       |
+| 10001 | Ray Client          |
+| 8000  | Ray Serve           |
+| 52365 | Ray Dashboard Agent |
+
+Get the RayService information with your RayService name.
+```shell
+$ kubectl describe  rayservices rayservice-sample
+```
+
+### Access User Services
+
+The users' traffic can go through the `serve` service (for example, `rayservice-sample-serve-svc`).
+
+#### Run a curl pod
+`kubectl run curl --image=radial/busyboxplus:curl -i --tty`
+
+For the fruit example deployment, you can try the following request
+```shell
+[ root@curl:/ ]$ curl  -X POST -H 'Content-Type: application/json' rayservice-sample-serve-svc.default.svc.cluster.local:8000 -d '["MANGO", 2]'
+6
+```
+You can get the response as `6`.
```

Comment:
During the in-person bug bash, the first command failed with `Error from server (AlreadyExists): pods "curl" already exists` and the second command failed with `curl: (6) Could not resolve host: t-rayservice-sample-serve-svc.default.svc.cluster.local`.  

The recommendation during the bug bash was to set up port forwarding.  If this is necessary, it should be included in the doc before this step.  But if it was some problem specific to the bug bash setup (like maybe the first command needed to succeed, but multiple users were sharing the `curl` pod) or user error, then no changes are needed

Reply:
The second is because the service is not created yet due to waiting for dashboard agent ready.